tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAwsVersioning.js

```diff
@@ -0,0 +1,548 @@
+            const key = `somekey-${Date.now()}`;
+            async.waterfall([
+                next => putToAwsBackend(s3, bucket, key, someBody,
+                    err => next(err)),
+                next => delAndAssertResult(s3, { bucket, key,
+                    resultType: nonVersionedDelete }, err => next(err)),
+                next => _getAssertDeleted(s3, { key, errorCode: 'NoSuchKey' },
+                    next),
+                next => _awsGetAssertDeleted({ key, errorCode: 'NoSuchKey' },
+                    next),
+            ], done);
+        });
+
+        it('versioning suspended: should create a delete marker in s3 ' +
+        'and aws successfully when deleting existing object', done => {
+            const key = `somekey-${Date.now()}`;
+            async.waterfall([
+                next => putNullVersionsToAws(s3, bucket, key, [someBody],
+                    err => next(err)),
+                next => delAndAssertResult(s3, { bucket, key, resultType:
+                    newDeleteMarker }, err => next(err)),
+                next => _getAssertDeleted(s3, { key, errorCode: 'NoSuchKey' },
+                    next),
+                next => _awsGetAssertDeleted({ key, errorCode: 'NoSuchKey' },
+                    next),
+            ], done);
+        });
+
+        it('versioning suspended: creating a delete marker should overwrite ' +
+        'existing null version (NoSuchObject even if delete marker is deleted)',
+        done => {
+            const key = `somekey-${Date.now()}`;
+            async.waterfall([
+                next => putNullVersionsToAws(s3, bucket, key, [undefined],
+                    err => next(err)),
+                // this delete marker should overwrite null version in s3
+                // metadata, while two delete markers are generated in aws
+                // (one to "delete" the prior "null" version and one is the
+                // new delete marker)
+                next => delAndAssertResult(s3, { bucket, key, resultType:
+                    newDeleteMarker }, next),
+                // delete delete marker
+                (dmVid, next) => delAndAssertResult(s3, { bucket, key,
+                    versionId: dmVid, resultType: deleteDeleteMarker },
+                    err => next(err)),
+                // should get no such object even after deleting del marker
+                next => _getAssertDeleted(s3, { key, errorCode: 'NoSuchKey' },
+                    next),
+                // in aws we should still get another delete marker instead
+                // of the null version
+                next => _awsGetAssertDeleted({ key, errorCode: 'NoSuchKey' },
+                    next),
+            ], done);
+        });
+
+        it('versioning suspended: creating a delete marker should overwrite ' +
+        'existing null version that is not the latest version in s3 md ' +
+        '(but not in AWS due to edge case)',
```

Comment:
by creating a delete marker, do you also mean just deleting the key on our s3?

Reply:
as discussed, you can create a delete marker while versioning is suspended and it will overwrite the latest null version.  please add a comment.